Duplicaat GDS2 berichten worden niet correct afgehandeld

### Probleem
Als er eerst berichten zijn geladen uit een bestanden stroom, bijv. met directory scanner en vervolgens worden berichten opgehaald vanuit GDS2 dan mislukt de duplicaat verwerking met een `detached entity passed to persist` fout.

De duplicaat controle is op laadProces nivo, dus de insert in de Bericht tabel mislukt, dat is verwacht, echter het bijwerken en opslaan van het laadproces mislukt en het ophalen stopt.

zie ook: 
- #439 
- mantis-15479

### Versie
versie info: <!-- ga naar http://<server:poort>/brmo-service/about.jsp en kopier/plak -->2.0.2

### Server log
<details><summary>server log:</summary>

```

2020-09-15 09:31:26 INFO  nl.b3p.brmo.service.scanner.GDS2OphalenProces#addLog - Downloaden https://service30.kadaster.nl/gds2/download/private/5c2fb0c9-00ad-4a97-98b1-107771147c08
2020-09-15 09:31:26 WARN  org.hibernate.engine.jdbc.spi.SqlExceptionHelper#logExceptions - SQL Error: 0, SQLState: 23505
2020-09-15 09:31:26 ERROR org.hibernate.engine.jdbc.spi.SqlExceptionHelper#logExceptions - ERROR: duplicate key value violates unique constraint "idx_bericht_refiddatumnr"
  Detail: Key (object_ref, datum, volgordenummer)=(NL.KAD.OnroerendeZaak:45150641670000, 2020-06-04 00:00:00, 1) already exists.
2020-09-15 09:31:26 WARN  nl.b3p.brmo.service.scanner.GDS2OphalenProces#laadAfgifte - Opslaan van bericht uit laadproces BKE-MUTBX01-STR03C6416-20200604-1.zip (5c2fb0c9-00ad-4a97-98b1-107771147c08) is mislukt.
javax.persistence.PersistenceException: org.hibernate.exception.ConstraintViolationException: could not execute statement
	at org.hibernate.internal.ExceptionConverterImpl.convert(ExceptionConverterImpl.java:154)
	at org.hibernate.internal.ExceptionConverterImpl.convert(ExceptionConverterImpl.java:181)
	at org.hibernate.internal.ExceptionConverterImpl.convert(ExceptionConverterImpl.java:188)
	at org.hibernate.internal.SessionImpl.firePersist(SessionImpl.java:724)
	at org.hibernate.internal.SessionImpl.persist(SessionImpl.java:704)
	at nl.b3p.brmo.service.scanner.GDS2OphalenProces.laadAfgifte(GDS2OphalenProces.java:703)
	at nl.b3p.brmo.service.scanner.GDS2OphalenProces.verwerkAfgiftes(GDS2OphalenProces.java:804)
	at nl.b3p.brmo.service.scanner.GDS2OphalenProces.execute(GDS2OphalenProces.java:435)
	at nl.b3p.brmo.service.stripes.GDS2OphalenUitvoerActionBean.execute(GDS2OphalenUitvoerActionBean.java:132)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at net.sourceforge.stripes.controller.DispatcherHelper$6.intercept(DispatcherHelper.java:456)
	at net.sourceforge.stripes.controller.ExecutionContext.proceed(ExecutionContext.java:176)
	at net.sourceforge.stripes.controller.BeforeAfterMethodInterceptor.intercept(BeforeAfterMethodInterceptor.java:113)
	at net.sourceforge.stripes.controller.ExecutionContext.proceed(ExecutionContext.java:173)
	at org.stripesstuff.plugin.waitpage.WaitPageInterceptor.intercept(WaitPageInterceptor.java:125)
	at net.sourceforge.stripes.controller.ExecutionContext.proceed(ExecutionContext.java:173)
	at net.sourceforge.stripes.controller.ExecutionContext.wrap(ExecutionContext.java:86)
	at net.sourceforge.stripes.controller.DispatcherHelper.invokeEventHandler(DispatcherHelper.java:454)
	at net.sourceforge.stripes.controller.DispatcherServlet.invokeEventHandler(DispatcherServlet.java:278)
	at net.sourceforge.stripes.controller.DispatcherServlet.service(DispatcherServlet.java:160)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:729)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:292)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)
	at net.sourceforge.stripes.controller.StripesFilter.doFilter(StripesFilter.java:260)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:240)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:240)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:212)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:94)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:616)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:141)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:79)
	at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:620)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:88)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:502)
	at org.apache.coyote.http11.AbstractHttp11Processor.process(AbstractHttp11Processor.java:1132)
	at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:684)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1539)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.run(NioEndpoint.java:1495)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.lang.Thread.run(Unknown Source)
Caused by: org.hibernate.exception.ConstraintViolationException: could not execute statement
	at org.hibernate.exception.internal.SQLStateConversionDelegate.convert(SQLStateConversionDelegate.java:109)
	at org.hibernate.exception.internal.StandardSQLExceptionConverter.convert(StandardSQLExceptionConverter.java:42)
	at org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:113)
	at org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:99)
	at org.hibernate.engine.jdbc.internal.ResultSetReturnImpl.executeUpdate(ResultSetReturnImpl.java:200)
	at org.hibernate.id.insert.AbstractSelectingDelegate.performInsert(AbstractSelectingDelegate.java:45)
	at org.hibernate.persister.entity.AbstractEntityPersister.insert(AbstractEntityPersister.java:3152)
	at org.hibernate.persister.entity.AbstractEntityPersister.insert(AbstractEntityPersister.java:3756)
	at org.hibernate.action.internal.EntityIdentityInsertAction.execute(EntityIdentityInsertAction.java:84)
	at org.hibernate.engine.spi.ActionQueue.execute(ActionQueue.java:645)
	at org.hibernate.engine.spi.ActionQueue.addResolvedEntityInsertAction(ActionQueue.java:282)
	at org.hibernate.engine.spi.ActionQueue.addInsertAction(ActionQueue.java:263)
	at org.hibernate.engine.spi.ActionQueue.addAction(ActionQueue.java:317)
	at org.hibernate.event.internal.AbstractSaveEventListener.addInsertAction(AbstractSaveEventListener.java:330)
	at org.hibernate.event.internal.AbstractSaveEventListener.performSaveOrReplicate(AbstractSaveEventListener.java:287)
	at org.hibernate.event.internal.AbstractSaveEventListener.performSave(AbstractSaveEventListener.java:193)
	at org.hibernate.event.internal.AbstractSaveEventListener.saveWithGeneratedId(AbstractSaveEventListener.java:123)
	at org.hibernate.event.internal.DefaultPersistEventListener.entityIsTransient(DefaultPersistEventListener.java:185)
	at org.hibernate.event.internal.DefaultPersistEventListener.onPersist(DefaultPersistEventListener.java:128)
	at org.hibernate.event.internal.DefaultPersistEventListener.onPersist(DefaultPersistEventListener.java:55)
	at org.hibernate.event.service.internal.EventListenerGroupImpl.fireEventOnEachListener(EventListenerGroupImpl.java:102)
	at org.hibernate.internal.SessionImpl.firePersist(SessionImpl.java:718)
	... 44 more
Caused by: org.postgresql.util.PSQLException: ERROR: duplicate key value violates unique constraint "idx_bericht_refiddatumnr"
  Detail: Key (object_ref, datum, volgordenummer)=(NL.KAD.OnroerendeZaak:45150641670000, 2020-06-04 00:00:00, 1) already exists.
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2440)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2183)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:308)
	at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:441)
	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:365)
	at org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:143)
	at org.postgresql.jdbc.PgPreparedStatement.executeUpdate(PgPreparedStatement.java:120)
	at org.apache.tomcat.dbcp.dbcp2.DelegatingPreparedStatement.executeUpdate(DelegatingPreparedStatement.java:97)
	at org.apache.tomcat.dbcp.dbcp2.DelegatingPreparedStatement.executeUpdate(DelegatingPreparedStatement.java:97)
	at org.hibernate.engine.jdbc.internal.ResultSetReturnImpl.executeUpdate(ResultSetReturnImpl.java:197)
	... 61 more
2020-09-15 09:31:26 WARN  nl.b3p.brmo.service.scanner.GDS2OphalenProces#laadAfgifte - Duplicaat bericht: NL.KAD.OnroerendeZaak:45150641670000:<?xml version="1.0" encoding="UTF-8"?>
.... BERICHT VERWIJDERD ....
2020-09-15 09:31:26 ERROR nl.b3p.brmo.service.scanner.GDS2OphalenProces#execute - Fout bij ophalen van GDS2 berichten
javax.persistence.PersistenceException: org.hibernate.PersistentObjectException: detached entity passed to persist: nl.b3p.brmo.persistence.staging.AutomatischProces
	at org.hibernate.internal.ExceptionConverterImpl.convert(ExceptionConverterImpl.java:154)
	at org.hibernate.internal.ExceptionConverterImpl.convert(ExceptionConverterImpl.java:181)
	at org.hibernate.internal.ExceptionConverterImpl.convert(ExceptionConverterImpl.java:188)
	at org.hibernate.internal.SessionImpl.firePersist(SessionImpl.java:748)
	at org.hibernate.internal.SessionImpl.persist(SessionImpl.java:710)
	at org.hibernate.engine.spi.CascadingActions$7.cascade(CascadingActions.java:298)
	at org.hibernate.engine.internal.Cascade.cascadeToOne(Cascade.java:492)
	at org.hibernate.engine.internal.Cascade.cascadeAssociation(Cascade.java:416)
	at org.hibernate.engine.internal.Cascade.cascadeProperty(Cascade.java:218)
	at org.hibernate.engine.internal.Cascade.cascade(Cascade.java:151)
	at org.hibernate.event.internal.AbstractSaveEventListener.cascadeBeforeSave(AbstractSaveEventListener.java:427)
	at org.hibernate.event.internal.AbstractSaveEventListener.performSaveOrReplicate(AbstractSaveEventListener.java:264)
	at org.hibernate.event.internal.AbstractSaveEventListener.performSave(AbstractSaveEventListener.java:193)
	at org.hibernate.event.internal.AbstractSaveEventListener.saveWithGeneratedId(AbstractSaveEventListener.java:123)
	at org.hibernate.event.internal.DefaultPersistEventListener.entityIsTransient(DefaultPersistEventListener.java:185)
	at org.hibernate.event.internal.DefaultPersistEventListener.onPersist(DefaultPersistEventListener.java:128)
	at org.hibernate.event.internal.DefaultPersistEventListener.onPersist(DefaultPersistEventListener.java:55)
	at org.hibernate.event.service.internal.EventListenerGroupImpl.fireEventOnEachListener(EventListenerGroupImpl.java:102)
	at org.hibernate.internal.SessionImpl.firePersist(SessionImpl.java:718)
	at org.hibernate.internal.SessionImpl.persist(SessionImpl.java:704)
	at nl.b3p.brmo.service.scanner.GDS2OphalenProces.laadAfgifte(GDS2OphalenProces.java:720)
	at nl.b3p.brmo.service.scanner.GDS2OphalenProces.verwerkAfgiftes(GDS2OphalenProces.java:804)
	at nl.b3p.brmo.service.scanner.GDS2OphalenProces.execute(GDS2OphalenProces.java:435)
	at nl.b3p.brmo.service.stripes.GDS2OphalenUitvoerActionBean.execute(GDS2OphalenUitvoerActionBean.java:132)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at net.sourceforge.stripes.controller.DispatcherHelper$6.intercept(DispatcherHelper.java:456)
	at net.sourceforge.stripes.controller.ExecutionContext.proceed(ExecutionContext.java:176)
	at net.sourceforge.stripes.controller.BeforeAfterMethodInterceptor.intercept(BeforeAfterMethodInterceptor.java:113)
	at net.sourceforge.stripes.controller.ExecutionContext.proceed(ExecutionContext.java:173)
	at org.stripesstuff.plugin.waitpage.WaitPageInterceptor.intercept(WaitPageInterceptor.java:125)
	at net.sourceforge.stripes.controller.ExecutionContext.proceed(ExecutionContext.java:173)
	at net.sourceforge.stripes.controller.ExecutionContext.wrap(ExecutionContext.java:86)
	at net.sourceforge.stripes.controller.DispatcherHelper.invokeEventHandler(DispatcherHelper.java:454)
	at net.sourceforge.stripes.controller.DispatcherServlet.invokeEventHandler(DispatcherServlet.java:278)
	at net.sourceforge.stripes.controller.DispatcherServlet.service(DispatcherServlet.java:160)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:729)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:292)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)
	at net.sourceforge.stripes.controller.StripesFilter.doFilter(StripesFilter.java:260)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:240)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:240)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:212)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:94)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:616)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:141)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:79)
	at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:620)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:88)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:502)
	at org.apache.coyote.http11.AbstractHttp11Processor.process(AbstractHttp11Processor.java:1132)
	at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:684)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1539)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.run(NioEndpoint.java:1495)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.lang.Thread.run(Unknown Source)
Caused by: org.hibernate.PersistentObjectException: detached entity passed to persist: nl.b3p.brmo.persistence.staging.AutomatischProces
	at org.hibernate.event.internal.DefaultPersistEventListener.onPersist(DefaultPersistEventListener.java:120)
	at org.hibernate.event.service.internal.EventListenerGroupImpl.fireEventOnEachListener(EventListenerGroupImpl.java:113)
	at org.hibernate.internal.SessionImpl.firePersist(SessionImpl.java:742)
	... 59 more



```

</details>

--- a/brmo-service/src/main/java/nl/b3p/brmo/service/scanner/GDS2OphalenProces.java
+++ b/brmo-service/src/main/java/nl/b3p/brmo/service/scanner/GDS2OphalenProces.java
@@ -638,6 +638,7 @@ public class GDS2OphalenProces extends AbstractExecutableProces {
         }
         lp.setSoort("brk");
         lp.setStatus(LaadProces.STATUS.STAGING_OK);
+        lp.setStatus_datum(new Date());
         SimpleDateFormat sdf = new SimpleDateFormat("dd-MM-yyyy HH:mm:ss");
         lp.setOpmerking("GDS2 download van " + url + " op " + sdf.format(new Date()));
         lp.setAutomatischProces(em.find(AutomatischProces.class, config.getId()));

--- a/brmo-service/src/main/java/nl/b3p/brmo/service/scanner/GDS2OphalenProces.java
+++ b/brmo-service/src/main/java/nl/b3p/brmo/service/scanner/GDS2OphalenProces.java
@@ -488,9 +488,7 @@ public class GDS2OphalenProces extends AbstractExecutableProces {
      * @param a de afgifte uit het soap verzoek
      * @param url te downloaden bestand
      * @throws Exception if any
-     * @see
-     * #laadAfgifte(nl.kadaster.schemas.gds2.afgifte_bestandenlijstgbresultaat.afgifte.v20130701.AfgifteGBType,
-     * java.lang.String)
+     * @see #laadAfgifte(AfgifteType, String)
      */
     private void laadBagAfgifte(AfgifteType a, String url) throws Exception {
         String msg = "Downloaden " + url;
@@ -714,14 +712,15 @@ public class GDS2OphalenProces extends AbstractExecutableProces {
             lp.setId(null);
             log.warn("Opslaan van bericht uit laadproces " + lp.getBestand_naam() + " is mislukt.", pe);
             log.warn("Duplicaat bericht: " + b.getObject_ref() + ":" + b.getBr_orgineel_xml() + "(" + b.getBr_xml() + ")");
+            lp.setOpmerking(lp.getOpmerking() + ": Fout, duplicaat bericht. Berichtinhoud: \n\n" + b.getBr_xml());
+            b = null;
             lp.setStatus(LaadProces.STATUS.STAGING_DUPLICAAT);
-            lp.setOpmerking(lp.getOpmerking() + ": Fout, duplicaat bericht. Inhoud: \n" + b.getBr_xml());
+            lp.setAutomatischProces(em.find(AutomatischProces.class, this.config.getId()));
             em.merge(this.config);
             em.persist(lp);
             em.flush();
             em.getTransaction().commit();
             em.clear();
-            b = null;
         }
         return b;
     }

--- a/pom.xml
+++ b/pom.xml
@@ -631,7 +631,7 @@
                 </plugin>
                 <plugin>
                     <artifactId>maven-release-plugin</artifactId>
-                    <version>2.5.3</version>
+                    <version>3.0.0-M1</version>
                 </plugin>
                 <plugin>
                     <artifactId>maven-install-plugin</artifactId>

--- a/pom.xml
+++ b/pom.xml
@@ -327,7 +327,7 @@
             <dependency>
                 <groupId>nl.b3p</groupId>
                 <artifactId>kadaster-gds2</artifactId>
-                <version>2.2</version>
+                <version>2.3</version>
             </dependency>
             <!-- J2EE dependencies en stripes -->
             <dependency>

